### DISCUSSION TITLE
How to print the pdf on button click

### DISCUSSION THREAD

**Discussion by Sharanravindran** [diegomura/react-pdf#2052] [General]
Hi right now, the only ways to print the pdf is to either click on the print option in the PDFviewer toolbar, the other way is to right click on the pdf and select print.

But what I want, is to print the pdf with a click on a button, is this possible?

Please show solution with React

**Comment by CoskunCakir** [comment]
Hi @Sharanravindran,

if you are using PDFviewer and since its an iframe you could do something like below

get the iframe element from the DOM
focus and print
assumed you have only one iframe in your document

```jsx
  const handlePrint = () => {
    const iframe: HTMLIFrameElement = document.getElementsByTagName('iframe')[0]
    iframe?.focus()
    iframe?.contentWindow?.print()
  }
```



let me know how it goes,
good luck
